book_source/index.Rmd

```diff
@@ -12,7 +12,7 @@ author: "By: PEcAn Team"
 
 **Ecosystem science, policy, and management informed by the best available data and models**
 
-```{r, echo=FALSE,out.height= "50%", out.width="50%", fig.align='center'}
+```{r, out.height= "50%", out.width="50%", fig.align='center'}
```

Comment:
I see that now. Is there somewhere where `echo=FALSE` is set more globally? I thought this could be done in the index. Otherwise, I don't know why code chunks aren't showing up elsewhere in the documentation.

Reply:
No it's set per chunk. By default it's TRUE

Reply:
Okay, well I don't have any ideas for fixing this then, but not having the R code show up makes using the vignettes in particular difficult.

Reply:
I think it's going to be a specific fix to that file then. So for the part that you noticed, for example this line: https://github.com/PecanProject/pecan/blob/develop/documentation/tutorials/MCMC/MCMC_Concepts.Rmd#L68

It could be that the `eval= FALSE` makes it ignore that section all together, so adding an explicit `echo=TRUE` would make it show up.

Reply:
I thought that might be the case, I was hoping for a more general solution. If you can confirm that adding the echo argument works, I'm happy to put them all in.

Reply:
Quickly chiming in.

You can set chunk defaults globally with code like the following:

```r
knitr::opts_chunk$set(message = FALSE, warnings = FALSE, echo =TRUE)
```

However, if you are compiling multiple Rmd files in one go, those defaults may get propagated through to other files that do not expect them.

Evidently, you can retrieve the default chunk options with the following:

```r
knitr::opts_chunk$get(default = TRUE)
```

...so I think it would be a good idea to add the following code for resetting chunk options to their defaults to the end of any Rmd files that set global chunk options in this way:

```r
knitr::opts_chunk$set(knitr::opts_chunk$get(default = TRUE))
```

@KristinaRiemer You could also add that line in an invisible code chunk (knitr chunk option `include = FALSE`) at the start of the photosynthesis vignette to ensure that the chunk options are set to their defaults (including `echo = TRUE`) before anything else happens.